ui/components/data-sources/BaseDataSourceDialog.js

```diff
@@ -200,7 +195,17 @@ class BaseDataSourceDialog extends Component {
     render() {
         const { visible } = this.props;
         const { name, type, err, success, validations } = this.state;
-        const submitButtonDisabled = some(validations, s => !s || s === "error");
+
+        // InMemory options are always correct: there is only one options which
```

Comment:
This validation logic belongs to `onTypeChange`.

Reply:
Won't work. We want the submit button for `InMemory` to be enabled by default without changing anything first (because there are no options to check: all options are always correct for in mem).